File: examples/unreliable_service.py
"""Example service with reliability risks for scanner demos."""

import asyncio

import requests
from fastapi import FastAPI
from sqlalchemy import create_engine

app = FastAPI()
engine = create_engine("postgresql://scanner:password@localhost/app")
jobs = asyncio.Queue()
DEBUG = True
REQUEST_TIMEOUT = 0


@app.post("/payments")
def create_payment():
    """Mutating endpoint with several production-readiness gaps."""
    response = requests.post("https://payments.example/charge")
    return response.json()


def worker():
    """Long-running loop without graceful shutdown handling."""
    while True:
        try:
            process_next_job()
        except Exception:
            pass

File: examples/unreliable_service.js
/**
 * Example JavaScript service with reliability risks for scanner demos.
 */

const express = require('express');
const axios = require('axios');
const { Pool } = require('pg');

const app = express();
const pool = new Pool({ connectionString: process.env.DATABASE_URL });

app.post('/orders', async (req, res) => {
  const payment = await axios.post('https://payments.example/charge', req.body);
  res.json(payment.data);
});

setInterval(async () => {
  try {
    await publishNextJob();
  } catch (err) {}
}, 1000);

module.exports = app;
